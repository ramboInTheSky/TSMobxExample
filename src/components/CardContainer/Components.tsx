import styled from '@emotion/styled'
import { Paper, IconButton } from '@material-ui/core'

import { variables } from '../../constants/theme'

export const Wrapper: React.ComponentType<any> = styled(Paper)`
    margin-top: ${variables.gutter};
    background-color: ${variables.componentBackgroundClear} !important;
`
export const Header = styled.div`
    background-color: ${variables.componentBackground};
    color: #333;
    font-size: 1.3rem;
    height: 25px;
    padding: ${variables.indentation};
    display: flex;
    justify-content: space-between;
`

export const Content: React.ComponentType<any> = styled.div`
    background-color: ${variables.componentBackgroundClear} !important;
    padding: ${variables.indentation};
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
`

export const Item: React.ComponentType<any> = styled(Paper)`
    background-color: ${variables.componentBackground} !important;
    margin: ${variables.indentation};
    display: grid;
    grid-template-rows: 32% 50%;
    height: 5rem;
    padding: ${variables.indentation};
    min-width: 10rem;
    max-width: 14rem;
    & div:nth-child(1){
        display: flex;
        flex-direction: row-reverse;
        cursor: pointer;
        font-size: .4rem;
        & svg{
            opacity: .5;
        }
    }
    & div:nth-child(2){
        padding: 5px;
        position: relative;
        font-size: 1.2rem;
        text-align: right; 
        & svg{
            opacity: .2;
            top: -36px;
            left: 0;
            position: absolute;
            font-size: 6rem;
            color: #999;
        }
    }
    
`
export const MenuHeader = styled.div`
    width: 100%;
    background-color: ${variables.dangerColor};
    height: 2rem;
    padding: 1rem;
    color: white;
    font-weight: 700;
    font-size: 1.2rem;
    outline: none;
    & span{
        font-weight: 400;
    }
`

// export const MenuButton: React.ComponentType<any> = styled(IconButton)`

// `

