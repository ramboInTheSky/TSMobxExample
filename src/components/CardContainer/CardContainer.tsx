import React, { version } from 'react'
import VpnKeyIcon from '@material-ui/icons/VpnKey'
import MeetingRoomIcon from '@material-ui/icons/MeetingRoom'
import DeleteIcon from '@material-ui/icons/DeleteOutline'
import { Wrapper, Header, Content, Item, MenuHeader } from './Components'
import { variables } from '../../constants/theme'
import { Menu, MenuItem } from '@material-ui/core';
import { LockGroup } from '../../models/LockGroups';

export interface CardContainerProps {
	type: 'rooms' | 'tags'
	items?: Array<string | LockGroup> 
	removeItemFn?: Function
}

export interface CardContainerState {
	menuOpen: { [key: string]: boolean }
}

export class CardContainer extends React.Component<CardContainerProps, CardContainerState> {
	constructor(props: CardContainerProps) {
		super(props)
		this.state = {
			menuOpen: this.initializePopups()
		}
	}

	initializePopups() {
		let obj = {}
		if (this.props.items) {
			this.props.items.forEach((item: any) => {
				obj[item.name || item] = false
			})
			return obj
		}
		else {
			return {}
		}
	}

	toggleMenu = (e: any, item: string) => this.setState({ menuOpen: { ...this.state.menuOpen, [item]: !this.state.menuOpen[item] } })

	render() {
		const { items, type } = this.props
		const { menuOpen } = this.state
		
		return items ? <Wrapper>
			<Header style={{ backgroundColor: `${type === 'tags' ? variables.tagColor : variables.roomColor}` }}>
				{type.toUpperCase()}
			</Header>
			<Content>
				{(items as any).map((_item: any, i: number) => {
					const item = _item.name || _item
					return <div id={`${type}-anchor-${item.replace(/ /g, '-')}`}>
						<Item key={item} >
							<div><DeleteIcon onClick={(e: any) => this.toggleMenu(e, item)} /></div>
							<div>{type === 'tags' ? <VpnKeyIcon /> : <MeetingRoomIcon />} {item}</div>

						</Item>
						<Menu anchorEl={document.querySelector(`#${type}-anchor-${item.replace(/ /g, '-')}`) as any} aria-label="Settings" id="Settings" title="Settings" open={menuOpen[item]} onClose={(e: any) => this.toggleMenu(e, item)}>
							<MenuHeader>Delete <span>{type === 'tags' ? 'Tag: ' : 'Room: '}{item}</span></MenuHeader>
							<MenuItem onClick={(e: any) => this.toggleMenu(e, item)}>Yes</MenuItem>
							<MenuItem onClick={(e: any) => this.toggleMenu(e, item)}>Cancel</MenuItem>
						</Menu>
					</div>
				})}
			</Content>
		</Wrapper> : null
	}
}
