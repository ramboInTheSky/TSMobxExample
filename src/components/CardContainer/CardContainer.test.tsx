import React from 'react'
import { shallow } from 'enzyme'

import { CardContainer } from './CardContainer'

describe('<CardContainer />', () => {
	let props: any
	beforeEach(()=>{
		props = {
			
		}
	})
	it('renders correctly', () => {
		expect(shallow(<CardContainer {...props}/>)).toMatchSnapshot()
	})

	// it('passes on props correctly', () => {
	// 	expect(shallow(<CardContainer foo="bar">Hello</CardContainer>)).toMatchSnapshot()
	// })

	// it('Does not re-render on children props change', () => {
	// 	const el: any = shallow(<CardContainer foo="bar" />)
	// 	el.setProps({ children: 'Goodbye' })
	// 	el.update()
	// 	expect(el).toMatchSnapshot()
	// })
})
