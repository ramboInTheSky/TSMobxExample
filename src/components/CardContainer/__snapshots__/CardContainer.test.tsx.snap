// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContainer /> renders correctly 1`] = `""`;
